numba/core/callconv.py

```diff
@@ -561,7 +568,18 @@ def call_function(self, builder, callee, resty, argtys, args):
         arginfo = self._get_arg_packer(argtys)
         args = list(arginfo.as_arguments(builder, args))
         realargs = [retvaltmp, excinfoptr] + args
-        code = builder.call(callee, realargs)
+        # deal with attrs, it's find to specify a load in a string like
+        # "noinline fast" as per LLVM or equally as an iterable of individual
+        # attributes.
+        if attrs is None:
+            _attrs = ()
+        elif isinstance(attrs, str):
+            _attrs = tuple(attrs.split(' '))
+        elif isinstance(attrs, Iterable):
```

Comment:
This is never tested.

Reply:
I can remove, though it just feels a bit strange API wise writing:
```
call_function(self, builder, callee, resty, argtys, args, attrs="noinline nounwind"):
```
vs.
```
call_function(self, builder, callee, resty, argtys, args, attrs=["noinline", "nounwind"]):
```
but I think testing the latter might be hard.

Reply:
Agree that the latter is nicer. And, I'd rather we have only one way of spelling it, even if we have to say `attrs=['noinline']`

Reply:
I've made a change along these lines, see what you think.